File: PlacementApp/frontend/src/services/skillService.js
import api from './api';

export const skillService = {
  getAllSkills: () => api.get('/skills'),
  getSkillById: (id) => api.get(`/skills/${id}`),
  getSkillsByCategory: (category) => api.get(`/skills/category/${category}`),
  addSkill: (skill) => api.post('/skills', skill),
  deleteSkill: (id) => api.delete(`/skills/${id}`),
};

File: PlacementApp/src/main/java/com/example/demo/service/SkillService.java
package com.example.demo.service;

import com.example.demo.entity.Skill;
import com.example.demo.repo.SkillRepo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
public class SkillService {
    @Autowired private SkillRepo skillRepo;

    public Skill addSkill(Skill skill) { return skillRepo.save(skill); }
    public List<Skill> getAllSkills() { return skillRepo.findAll(); }
    public Skill getSkillById(Long id) { return skillRepo.findById(id).orElseThrow(); }
    public List<Skill> getSkillsByCategory(String category) { return skillRepo.findByCategory(category); }
    public void deleteSkill(Long id) { skillRepo.deleteById(id); }
}
